File: src/brs/assetexchange/AssetAccountServiceImpl.java
package brs.assetexchange;

import brs.Account.AccountAsset;
import brs.Asset;
import brs.db.store.AccountStore;

import java.util.Collection;

class AssetAccountServiceImpl {

  private final AccountStore accountStore;

  public AssetAccountServiceImpl(AccountStore accountStore) {
    this.accountStore = accountStore;
  }

  public Collection<AccountAsset> getAssetAccounts(Asset asset, boolean ignoreTreasury, long minimumQuantity, int from, int to) {
    return accountStore.getAssetAccounts(asset, ignoreTreasury, minimumQuantity, from, to);
  }

  public int getAssetAccountsCount(Asset asset, long minimumQuantity, boolean ignoreTreasury) {
    return accountStore.getAssetAccountsCount(asset, minimumQuantity, ignoreTreasury);
  }
  
  public long getAssetCirculatingSupply(Asset asset) {
    return accountStore.getAssetCirculatingSupply(asset);
  }

}

File: src/main/java/brs/assetexchange/AssetAccountServiceImpl.kt
package brs.assetexchange

import brs.Account.AccountAsset
import brs.db.store.AccountStore

internal class AssetAccountServiceImpl(private val accountStore: AccountStore) { // TODO interface

    fun getAssetAccounts(assetId: Long, from: Int, to: Int): Collection<AccountAsset> {
        return accountStore.getAssetAccounts(assetId, from, to)
    }

    fun getAssetAccounts(assetId: Long, height: Int, from: Int, to: Int): Collection<AccountAsset> {
        return if (height < 0) {
            getAssetAccounts(assetId, from, to)
        } else accountStore.getAssetAccounts(assetId, height, from, to)
    }

    fun getAssetAccountsCount(assetId: Long): Int {
        return accountStore.getAssetAccountsCount(assetId)
    }
}
